app/controllers/v0/yellow_ribbon_programs_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module V0
+  class YellowRibbonProgramsController < ApiController
+    include Facets
+
+    # GET /v0/yellow_ribbon_programs?name=duluth
+    def index
+      @meta = {
+        version: @version,
+        count: search_results.count
+      }
+      render json: search_results.order('school_name_in_yr_database ASC NULLS LAST').page(params[:page]), meta: @meta
+    end
+
+    private
+
+    def normalized_query_params
+      query = params.deep_dup
+      query.tap do
+        query[:name].try(:strip!)
+        query[:name].try(:downcase!)
+      end
+    end
+
+    def search_results
+      @query ||= normalized_query_params
+      YellowRibbonProgramSource.search(@query[:name])
```

Comment:
I would recommend using https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/yellow_ribbon_program.rb instead as this table is linked to the relevant institution and if needed Ruby on Rails can be used to retreive institution data

Reply:
Hmm, `YellowRibbonProgramSource` seems to have exactly what I need atm 🤔

Reply:
The issue with using `yellow_ribbon_program_source` is that it is blown away every time a CSV is successfully uploaded meaning the data will change on the fly whenever someone updates the CSV instead of only when a “preview generation” and “publish to production” occur

Reply:
If `YellowRibbonProgram` is missing needed data create a [Active Record Migration](https://edgeguides.rubyonrails.org/active_record_migrations.html) while following [Strong Migrations Gem](https://github.com/ankane/strong_migrations) to add the missing columns to the table

Then in https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/institution_builder.rb#L449 update the SQL to insert the data from `yellow_ribbon_program_source` into `yellow_ribbon_program`

Reply:
Perfect, thanks for this suggestion! I updated the PR to only use `YellowRibbonProgram` instead of `YellowRibbonProgramSource` now 🙂